zokrates_core/src/flatten/mod.rs

```diff
@@ -349,12 +349,33 @@ impl Flattener {
 
                 FlatExpression::Identifier(name_1_y)
             },
+            BooleanExpression::AndAnd(box lhs, box rhs) => {
+                let x = self.flatten_boolean_expression(
+                    functions_flattened,
+                    arguments_flattened,
+                    statements_flattened,
+                    lhs
+                );
+                let y = self.flatten_boolean_expression(
+                    functions_flattened,
+                    arguments_flattened,
+                    statements_flattened,
+                    rhs
+                );
+                let name_new = format!("sym_{}", self.next_var_idx);
+
+                statements_flattened.push(FlatStatement::Definition(
+                    name_new.to_string(), FlatExpression::Mult(box x, box y))
```

Comment:
One thing to be careful about is that `FlatExpression`s are not necessarily linear, so multiplying two of them is not guaranteed to be quadratic. This is a bit awkward and may change in the future (using linear expressions as the main intermediate concept rather than quadratic now) but for now using `.is_linear()` is sometimes necessary to make sure we always end up with something quadratic in the flat program. Check out the flattening of `Mult` in `flatten_field_ewpression` for an approach, and feel free to add `assert(x.is_linear() && y.is_linear())` before pushing the definition

Reply:
Since x and y are flattened boolean expressions, and `faltten_boolean_expression` always returns an identifier, I believe they should always be linear. I've added an assertion to document that

Reply:
Great thank you, good to have for when `flatten_boolean_expression` starts returning something else.